### PR TITLE
Feat: [SRM-14043]: Adding permissions for downtime

### PR DESCRIPTION
**Pull request by Varsha-lalwani**
## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>

You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger utAll`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/44269)

### CODE REVIEW

**Comment by reetika04** [comment]
Please register the Resource in Resource Groups as well.

**Review by reetika04 — COMMENTED** [review]
> Reviewed 4 of 4 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/44269)* status: 0 of 1 approvals obtained.

**Review by deepak-harness — APPROVED** [review]
> Reviewed 4 of 4 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/44269)* status: :shipit: complete! ✓

**Review by deepak-harness — DISMISSED** [review]
> Reviewed 4 of 5 files at r2, 12 of 12 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/44269)* status: :shipit: complete! ✓

___
*[`srm-service/modules/cv-nextgen-service/service/src/main/java/io/harness/cvng/downtime/resources/DowntimeProjectLevelResource.java` line 204 at r3](https://reviewable.io/reviews/harness/harness-core/44269#-NOyTELg9JtvEIQ2ng_N:-NOyTELg9JtvEIQ2ng_O:b-vyrmex) ([raw file](https://github.com/harness/harness-core/blob/5214a307a52dbf05f2efaaacbab06b222d654e71/srm-service/modules/cv-nextgen-service/service/src/main/java/io/harness/cvng/downtime/resources/DowntimeProjectLevelResource.java#L204)):*
> ```Java
>         responses = { @io.swagger.v3.oas.annotations.responses.ApiResponse(description = "Enables or Disables Downtime")
>      })*/
>   @NGAccessControlCheck(resourceType = DOWNTIME, permission = VIEW_PERMISSION)
> ```

nit: change it to EDIT_PERMISSION in the next PR.

File: platform-service/modules/resource-group-service/src/main/java/io/harness/resourcegroup/resourceclient/downtime/DowntimeResourceImpl.java [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+  }
+
+  @Override
+  public List<Boolean> validate(List<String> resourceIds, Scope scope) {
+    return Collections.emptyList();
+  }
+
+  @Override
+  public Map<ScopeLevel, EnumSet<ValidatorType>> getSelectorKind() {
+    return ImmutableMap.of(ScopeLevel.ACCOUNT, EnumSet.of(BY_RESOURCE_TYPE_INCLUDING_CHILD_SCOPES),
```
Comment (reetika04) [inline platform-service/modules/resource-group-service/src/main/java/io/harness/resourcegroup/resourceclient/downtime/DowntimeResourceImpl.java:75]: Add BY_RESOURCE_TYPE as well.

**Review by reetika04 — COMMENTED** [review]
> Reviewed 3 of 5 files at r2, 1 of 12 files at r3, 11 of 12 files at r4, 1 of 1 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/44269)* status: 0 of 1 approvals obtained.

**Comment by Varsha-lalwani** [comment]
trigger ut1

**Comment by Varsha-lalwani** [comment]
trigger ut1

**Review by Jimit09Gandhi — APPROVED** [review]
> Reviewed all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/44269)* status: :shipit: complete! ✓

**Review by deepak-harness — APPROVED** [review]
> Reviewed 11 of 12 files at r4, 1 of 1 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/44269)* status: :shipit: complete! ✓

**Pull request merged**